src/main/java/io/hops/hopsworks/expat/migrations/featurestore/statistics/StatisticsMigration.java

```diff
@@ -772,39 +798,49 @@ private String statisticsFileName(Long startCommitTime, Long endCommitTime, Stri
     return name + endCommitTime + "_" + featureName + ".json";
   }
   
-  private boolean updateFeatureGroupStatisticsCommitWindow(PreparedStatement updateFgsStmt, Integer fgStatisticsId,
-    Long windowStartCommitTime) throws SQLException {
+  private boolean loadFeatureGroupStatisticsCommitWindow(Set<FeatureGroupStatisticsCommitWindow> loadList,
+      Integer fgStatisticsId, Long windowStartCommitTime) throws SQLException {
     if (windowStartCommitTime == null) {
       return false; // skip update, no time travel-enabled feature group
     }
-    updateFgsStmt.setLong(1, windowStartCommitTime);
-    updateFgsStmt.setInt(2, fgStatisticsId);
-    updateFgsStmt.addBatch();
+    loadList.add(new FeatureGroupStatisticsCommitWindow(fgStatisticsId, windowStartCommitTime));
     return true;
   }
-  
-  private void deleteFeatureGroupStatistics(PreparedStatement deleteFgsStmt, Integer fgStatisticsId)
-      throws SQLException {
-    deleteFgsStmt.setInt(1, fgStatisticsId);
-    deleteFgsStmt.addBatch();
-  }
-  
-  private void deleteFeatureGroupStatisticsActivity(PreparedStatement deleteFgsActStmt, Integer fgStatisticsId)
-    throws SQLException {
-    deleteFgsActStmt.setInt(1, fgStatisticsId);
-    deleteFgsActStmt.addBatch();
+
+  private void updateFeatureGroupStatisticsCommitWindow(PreparedStatement updateFgsStmt,
+      Set<FeatureGroupStatisticsCommitWindow> listToUpdate) throws SQLException {
+    int c = 0;
```

Comment:
```suggestion
    int c = 1;
```
Again harmless, but if we set c=0, the first iteration will run the update sql query for just one statistics id in the batch.